cmd/cometbft/commands/debug/kill.go

```diff
@@ -123,11 +124,15 @@ func killProc(pid uint64, dir string) error {
 	go func() {
 		// Killing the CometBFT process with the '-ABRT|-6' signal will result in
 		// a goroutine stacktrace.
-		p, err := os.FindProcess(int(pid))
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "failed to find PID to kill CometBFT process: %s", err)
-		} else if err = p.Signal(syscall.SIGABRT); err != nil {
-			fmt.Fprintf(os.Stderr, "failed to kill CometBFT process: %s", err)
+		if pid <= math.MaxInt {
```

Comment:
Honestly, I'm scratching my head at this. I think the killProc function should already only accept int, if processes are stored as int. Accepting uint64 because that's a better validation seems counterintuitive, if it is then converted again.

OS processes should always be bigger than 0 for sure, but the high limit is dependent on the architecture. (32-bit vs 64-bit)

If FindProcess requires an int, I would make killProc require an int too. No need to make it more complex.

Reply:
> I think the killProc function should already only accept int

TBH, I was hesitating whether to add the limit check, or go all the way until the CLI param... and went for the former. Main reason: limiting code changes on what will likely be backported to all branched.

Now that I see you're leaning towards the other solution, let's go for it. Bear with me